strategist/src/phases/settlement.rs

```
@@ -96,32 +95,75 @@ impl Strategy {
             info!(
                 target: SETTLEMENT_PHASE, "settling obligation #{}", obligation.id
             );
-            let obligation_settlement_msg = to_json_binary(
-                &valence_clearing_queue_supervaults::msg::FunctionMsgs::SettleNextObligation {},
-            )?;
 
-            self.enqueue_neutron(SETTLE_OBLIGATION_LABEL, vec![obligation_settlement_msg])
-                .await?;
+            // build the settlement function message
+            let settlement_exec_msg =
+                valence_library_utils::msg::ExecuteMsg::<_, ()>::ProcessFunction(
+                    valence_clearing_queue_supervaults::msg::FunctionMsgs::SettleNextObligation {},
+                );
+
+            // enqueue the settlement message and tick the processor
+            self.enqueue_neutron(
+                SETTLE_OBLIGATION_LABEL,
+                vec![to_json_binary(&settlement_exec_msg)?],
+            )
+            .await?;
 
             self.tick_neutron().await?;
         }
 
         Ok(())
     }
 }
+
+/// helper function that flattens a vec of withdraw obligations
+/// into a single batch.
+/// returns a tuple: (amount_1, amount_2), respecting the order
+/// of the (denom_1, denom_2) input
+fn flatten_obligation_queue_amounts(
+    obligations: &[WithdrawalObligation],
+    (denom_1, denom_2): (String, String),
+) -> (u128, u128) {
+    let mut amount_1 = 0;
+    let mut amount_2 = 0;
+
+    // iterate through all obligations and sum up the coin amounts
+    for withdraw_obligation in obligations.iter() {
+        for payout_coin in withdraw_obligation.payout_coins.iter() {
+            if payout_coin.denom == denom_1 {
+                amount_1 += payout_coin.amount.u128();
+            } else if payout_coin.denom == denom_2 {
+                amount_2 += payout_coin.amount.u128();
+            } else {
+                warn!(target: SETTLEMENT_PHASE, "obligation contains unrecognized denom: {}", payout_coin.denom);
```

should probably be a panic? 

This could happen if we didn't pay out obligations and did a phase shift right?

---

if it should never happen then we should panic as we're entering unknown territory